### PR TITLE
Add video list rendering (grid layout) with dummy data

### LINKED ISSUE #84: Render content-blocks
Blocked by #83. After that is merged, the frontend needs to request content blocks and render them accordingly.

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Only dummy data currently, but at least something one can use.

Screenshot from 2021-01-28 18-22-59

Currently, it's not particularly optimized for small screens, but it also doesn't break.

Closes #84

### CODE REVIEW

File: frontend/src/ui/Blocks.tsx [hunk 1]
```diff
@@ -57,12 +57,77 @@ const TextBlock: React.FC<TextProps> = ({ title, content }) => (
...
+                    return (
+                        <div
+                            key={v.id}
+                            css={{
+                                margin: "12px 8px",
+                                width: THUMB_WIDTH,
+                                "& a": { color: "black", textDecoration: "none" },
+                            }}
+                        >
+                            <a href={url} css={{ position: "relative", display: "block" }}>
```
Comment (JulianKniephoff) [inline frontend/src/ui/Blocks.tsx:82]: This should probably be a `<Link>` from `react-router`. An `a` reloads the entire page. I only think of this now, unfortunately; we should probably check existing code for this as well. :sweat_smile:

File: frontend/src/ui/Blocks.tsx [hunk 2]
```diff
@@ -57,12 +57,77 @@ const TextBlock: React.FC<TextProps> = ({ title, content }) => (
...
+        </Block>
+    );
+};
+
+const formatLength = (totalSeconds: number) => {
+    const seconds = totalSeconds % 60;
+    const minutes = Math.floor(totalSeconds / 60) % 60;
+    const hours = Math.floor(totalSeconds / (60 * 60));
+
+    const pad = (n: number) => (n < 10 ? `0${n}` : `${n}`);
```
Comment (JulianKniephoff) [inline frontend/src/ui/Blocks.tsx:121]: I don't mind using this specialized function, but I also don't want to fail to mention that there is actually `String.prototype.padStart` now. ^_^

File: frontend/src/ui/Blocks.tsx [hunk 3]
```diff
@@ -57,12 +57,77 @@ const TextBlock: React.FC<TextProps> = ({ title, content }) => (
...
+
+    if (hours > 0) {
+        return `${hours}:${pad(minutes)}:${pad(seconds)}`;
+    } else if (minutes > 0) {
+        return `${minutes}:${pad(seconds)}`;
+    } else {
+        return `0:${pad(seconds)}`;
```
Comment (JulianKniephoff) [inline frontend/src/ui/Blocks.tsx:128]:
```suggestion
    } else {
        return `${minutes}:${pad(seconds)}`;
```

Shouldn't this be the same (ignoring `minutes < 0`, which should not happen)? :thinking:

**Review by JulianKniephoff — CHANGES_REQUESTED** [review]
As usual only a few minor thing, one of which we need to check in the rest of the code base as well. :/

File: frontend/src/ui/Blocks.tsx (hunk 2 above)
Comment (LukasKalbertodt) [reply frontend/src/ui/Blocks.tsx:136]: Ui, good to know. But yeah, I guess our function is fine here.

File: frontend/src/ui/Blocks.tsx (hunk 3 above)
Comment (LukasKalbertodt) [reply frontend/src/ui/Blocks.tsx:143]: Ha, good point :D

File: frontend/src/ui/Blocks.tsx (hunk 1 above)
Comment (LukasKalbertodt) [reply frontend/src/ui/Blocks.tsx:97]: Yeah good catch. I checked the rest of the code base: it's fine everywhere.

**Comment by LukasKalbertodt** [comment]
Just for the protocol on why I added my own lecture as example data here: lecture recordings are a common video in our production systems, so it make sense to have some lecture recordings in our example data. And this was just the only lecture I could think of that I could easily use thumbnails of and link to directly without any license problems. I don't wanna self-promote :D if anyone has a better idea for example lecture data, let me know.

File: frontend/src/ui/Blocks.tsx (hunk 1 above)
Comment (JulianKniephoff) [reply frontend/src/ui/Blocks.tsx]: Ok phew great thx. :sweat_smile: :relieved:

**Review by JulianKniephoff — COMMENTED** [review]
Just two small things to think about, maybe. :)

File: backend/server/src/db/migrations/5-events.sql [hunk 4]
```diff
@@ -9,6 +9,7 @@ create table events (
     -- Meta data
     title text not null,
     description text,
+    duration int, -- in seconds
```
[the remark this answers is not in the document]
Comment (LukasKalbertodt) [reply backend/server/src/db/migrations/5-events.sql:12]: Oh yeah totally. Fixed (amended).

File: frontend/src/ui/Blocks.tsx [hunk 5]
```diff
@@ -23,6 +23,12 @@ export const Blocks: React.FC<Props> = ({ realm }) => {
                     title
                     __typename
                     ... on Text { content }
+                    ... on VideoList {
+                        series {
+                            title
+                            events { id title thumbnail duration }
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (LukasKalbertodt) [reply frontend/src/ui/Blocks.tsx:32]: I would suggest doing that later: https://github.com/elan-ev/tobira/issues/160

**Review by JulianKniephoff — APPROVED** [review]
Yeah ok cool, this is really coming to life! :smile:

**Pull request merged**